### DISCUSSION TITLE
Unable to add Java FX dependencies

### DISCUSSION THREAD

**Discussion by ajaychandran** [scalameta/metals#4701] [Q&A] [answered]
The current version of Coursier is unable to download [platform specific Java FX dependencies](https://github.com/coursier/coursier/pull/2176).

I [installed a development version](https://github.com/scalacenter/bloop/blob/3c35973b4f842c06da55ea01e1284cf21ca25d7f/website/content/docs/installation.md) of Bloop with an upgraded version of Coursier.

Is it possible to force Metals to use this external server?
Any help would be much appreciated. Thanks.

```sh
2022.12.05 01:48:32 INFO    not found: https://repo1.maven.org/maven2/org/openjfx/javafx-controls/11.0.2/javafx-controls-11.0.2-${javafx.platform}.jar
2022.12.05 01:48:32 INFO    not found: https://repo1.maven.org/maven2/org/openjfx/javafx-graphics/11.0.2/javafx-graphics-11.0.2-${javafx.platform}.jar
2022.12.05 01:48:32 INFO    not found: https://repo1.maven.org/maven2/org/openjfx/javafx-base/11.0.2/javafx-base-11.0.2-${javafx.platform}.jar
```

**Comment by tgodzik** [comment]
Thanks for the question! The errors seems to be coming from sbt. Bloop doesn't download dependencies besides a few needed for it's normal functioning. We do however have additional plugins for each build tool, mill and sbt use coursier from what I remember.

Could you give a bit more background about which tool you are using?

**Comment by ajaychandran** [comment]
I am using mill 0.10.9. Mill uses Coursier [2.1.0-RC1](https://github.co.m/com-lihaoyi/mill/blob/895f175851259d733896db4684082426d822a05e/build.sc#L145).

Bloop 1.5.4 uses Coursier [1.0.6](https://github.com/scalacenter/bloop/blob/41662d3384a5c0cff1be6379f79aef4799b11f6d/project/Dependencies.scala#L72).

**Reply by tgodzik** [reply]
Do you maybe hava a full log that was run during importing the project?

**Comment by ajaychandran** [comment, marked as answer]
The issue disappeared after clearing the `tmp` directory.
Thanks anyway.
